### PR TITLE
Edit tag format for update command

### PR DESCRIPTION
**Pull request by qlchan24**
- Multiple tags are now combined into a single prefix for Update Command, in line with find command.
- Update messages, documentation and diagrams with the updated command format for Update Command.
- Logic and Storage component class diagrams updated.
- Bug fixes

### CODE REVIEW

**Review by jeffreytjs — COMMENTED** [review]
LGTM!

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -1476,6 +1476,62 @@ testers are expected to do more *exploratory* testing.
 
    1. Other incorrect delete commands to try: `delete`, `delete x`, `...` (where x is larger than the list size)<br>
       Expected: Similar to previous.
+ 
+### Updating a Product in a Supplier/Warehouse
+
+1. Update command format: `update ct/TYPE n/NAME pd/PRODUCT_NAME [q/QUANTITY] [t/TAG…​]`
```
Comment (zhengweii) [inline docs/DeveloperGuide.md]: Oops prefix should be index instead of name here!

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -1476,6 +1476,62 @@ testers are expected to do more *exploratory* testing.
...
+      Expected: No macro created. Error details is shown in the displayed message.
+      
+### Removing a macro
+
+1. Remove macro command format: `removemacro ALIAS`
+
+   1. Prerequisites: At least one macro presently saved in the application. 
+
+   1. Test case: Alias exists in a saved macro e.g. `removemacro uw`<br>
+      Expected: The macro with the alias `uw` is removed. Details of the new macro is shown in the display message.
```
Comment (zhengweii) [inline docs/DeveloperGuide.md]: Hmm just wondering if "details of new macro" is wrong??

File: docs/DeveloperGuide.md [hunk 3]
```diff
@@ -1476,6 +1476,62 @@ testers are expected to do more *exploratory* testing.
...
+
+   1. Test case: Alias does not exist in any saved macro e.g. `removemacro a/magic`<br>
+      Expected: No macro removed. Error details is shown in the displayed message.
+
+### Assigning a macro
```
Comment (zhengweii) [inline docs/DeveloperGuide.md]: Should be list here!

File: src/main/java/seedu/clinic/commons/core/Messages.java [hunk 4]
```diff
@@ -7,6 +7,7 @@
 
     public static final String MESSAGE_UNKNOWN_COMMAND = "Unknown command";
     public static final String MESSAGE_INVALID_COMMAND_FORMAT = "Invalid command format! \n\n%1$s";
+    public static final String MESSAGE_INVALID_ALIAS = "Invalid Alias! \n\n%1$s";
```
Comment (zhengweii) [inline src/main/java/seedu/clinic/commons/core/Messages.java]: We might want to make "Alias" lowercase for consistency!

File: docs/DeveloperGuide.md (hunk 1 above)
Comment (qlchan24) [reply docs/DeveloperGuide.md]: Thanks! will fix

File: docs/DeveloperGuide.md (hunk 2 above)
Comment (qlchan24) [reply docs/DeveloperGuide.md]: Yes thats true. thanks!

File: docs/DeveloperGuide.md (hunk 3 above)
Comment (qlchan24) [reply docs/DeveloperGuide.md]: Yes thanks!

File: src/main/java/seedu/clinic/commons/core/Messages.java (hunk 4 above)
Comment (qlchan24) [reply src/main/java/seedu/clinic/commons/core/Messages.java]: Fair point. Thanks!

**Review by tohyuting — APPROVED** [review]
LGTM

**Pull request merged**